### ISSUE TITLE
Rule Change: check TypeScript signatures consistently in `max-params`

### ISSUE THREAD

**Issue by xbinaryx** [eslint/eslint#21336]
### What rule do you want to change?

`max-params`

### What change do you want to make?

Generate more warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```ts
/* eslint max-params: ["error", 3] */

// Reported today
declare function f(a: number, b: number, c: number, d: number): void;
type F = (a: number, b: number, c: number, d: number) => void;
interface I {
    prop: (a: number, b: number, c: number, d: number) => void;
}

// Not reported today
interface J {
    method(a: number, b: number, c: number, d: number): void;
    (a: number, b: number, c: number, d: number): void;
    new (a: number, b: number, c: number, d: number): J;
}
type C = new (a: number, b: number, c: number, d: number) => object;
declare class D {
    constructor(a: number, b: number, c: number, d: number);
    method(a: number, b: number, c: number, d: number): void;
}
abstract class E {
    abstract method(a: number, b: number, c: number, d: number): void;
}
```

### What does the rule currently do for this code?

Everything in the first block is reported, and nothing in the second block is reported.

### What will the rule do after it's changed?

Everything in the second block would be reported.

### Participation

- [x] I am willing to submit a pull request to implement this change.

### AI acknowledgment

- [x] I did not use AI to generate this issue report.
- [ ] (If the above is not checked) I have reviewed the AI-generated content before submitting.

**Comment by DMartens** [comment]
I agree that the rule should *optionally* have support for this as these are all function definitions.
If for example a class implements the interface with a method with too many parameters, the same error would also be reported there.
I could also not find discussion regarding this in `typescript-eslint`, neither in the original [PR](https://github.com/typescript-eslint/typescript-eslint/pull/7696) which only ignores the `this` parameter and the extension for the extension of [`TSDeclareFunction` and `TSFunctionType`](https://github.com/typescript-eslint/typescript-eslint/pull/10312)

**Comment by KaustAbhinand** [comment]
Hi has this issue been fixed? If not I would like to go ahead :)
